crates/ethereum/node/src/node.rs

```diff
@@ -233,8 +233,10 @@ where
         ctx: &BuilderContext<Node>,
         pool: Pool,
     ) -> eyre::Result<PayloadBuilderHandle<Types::Engine>> {
-        let payload_builder =
-            reth_ethereum_payload_builder::EthereumPayloadBuilder::new(self.evm_config);
+        let payload_builder = reth_ethereum_payload_builder::EthereumPayloadBuilder::new(
+            self.evm_config,
+            NoopTransactionFilter::new(),
+        );
```

Comment:
I know this isn't great and would probably be better to have the filter in `EthereumPayloadBuilder` here too, but this added a lot of constraints:
- either we added a generic `TxFilter` to it as done lower, which meant we add to add this generic to `EthereumNode::components`and the `Node` trait implementation of it.
- we have a default filter which would be `NoopTransactionFilter` but this filter requires a GAT `Transaction` which represents the transaction we are filtering over (in this case a pool transaction).

I feel like the second solution is better but not quite doable in a clean way yet, since I think we need `FullNodeTypes` to expose the pool transaction type.